templates/nextflow-aurora-mysql8.yaml

```diff
@@ -0,0 +1,461 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Serverless Aurora MySQL DB for NextFlow Tower
+
+Parameters:
+
+  VpcID:
+    Description: ID of VPC
+    Type: AWS::EC2::VPC::Id
+
+  SubnetIDs:
+    Description: A list of subnets in the chosen VPC
+    Type: List<AWS::EC2::Subnet::Id>
+
+  DBClusterIdentifier:
+    Description: Unique identifier of databse cluster.
+    Type: String
+    MinLength: 1
+    MaxLength: 63
+    AllowedPattern: "^(?!.*--)[a-zA-Z]+[0-9a-zA-Z-]*[0-9a-zA-Z]$"
+    ConstraintDescription: >
+      1 to 63 alphanumeric characters or hyphens.
+      First character must be a letter.
+      Can't contain two consecutive hyphens.
+      Can't end with a hyphen.
+    Default: tower
+
+  DBName:
+    Description: Optional. Database Name within cluster.
+    Type: String
+    MinLength: 1
+    MaxLength: 64
+    AllowedPattern: "^[a-zA-Z]+[0-9,a-z,A-Z$_]*$"
+    ConstraintDescription: >
+      1 to 64 alphanumeric characters, dollar sign, or underscore.
+      First character must be a letter.
+    Default: tower
+
+  DBUserName:
+    Description: Optional. Username to be created for database.
+    Type: String
+    MinLength: 1
+    MaxLength: 32
+    Default: tower
+
+  SnapshotName:
+    Description: Optional. Snapshot ID to restore database. Leave this blank if you are not restoring from a snapshot.
+    Type: String
+    Default: ''
```

Comment:
In the PR description you mention

> manually migrate the existing data from the old 5.7 DB to the new 8.0 DB then switch the Seqera app servers to use the new DB.

Is this `SnapshotName` where that manual data migration is going to happen, if it's possible to do a direct migration for a snapshot from 5.7 to 8.0?

Reply:
No, it's not possible.  This was designed for migrating data from the same db type.  Going from mysql 5.7 to 8.0 will result in the following error.. `AuroraCluster AWS::RDS::DBCluster CREATE_IN_PROGRESS The engine version you requested for your restored DB cluster (8.0.mysql_aurora.3.08.1) is not compatible with the engine version of the DB cluster snapshot (5.7.mysql_aurora.2.11.4).`